Feature/option to save a shared calendar

cristobalwalters12 opened the pull request:
Resolve #38 

- Created a component for displaying messages when a calendar is successfully saved.
- Implemented a button and associated methods for saving shared calendars.
- Developed a method and getter to check whether a calendar is saved locally or via the API, specifically within the user's account.
- If the calendar is already saved in the user's account, the "Save Calendar" button is not displayed.

cristobalwalters12 committed:
avance calendarios compartidos

cristobalwalters12 committed:
Avance 2

cristobalwalters12 committed:
avance 3

cristobalwalters12 committed:
Avance 4

- creación de componente de dialog
- modificaciones en el store de calendar
- modificaciones en calendarShow

cristobalwalters12 committed:
Merge branch 'master' of https://github.com/BaaltRodrigo/duocmatico into feature/Option-to-save-a-shared-calendar

cristobalwalters12 committed:
Add API call to get calendars after deleting a
calendar.

cristobalwalters12 committed:
Add calendarExists getter and update Guardar
Calendario button visibility

cristobalwalters12 committed:
Update DmCalendarMessage.vue with successful save
message

cristobalwalters12 committed:
Remove  error and success properties from state.

BaaltRodrigo commented on src/components/calendar/DmCalendarMessage.vue:
```diff
@@ -0,0 +1,23 @@
+<template>
+  <div class="text-center">
```
May the _text-center_ class be part of the `v-card`?

not very semantic the `div` tag

BaaltRodrigo commented on src/store/calendars.js:
```diff
@@ -130,23 +129,28 @@ const actions = {
+      ...calendar,
+      uuid: uuidv4(),
+    };
+    console.log(calendarData);
```
Remove `console.log` when doing a PR

BaaltRodrigo commented on src/views/CalendarShow.vue:
```diff
@@ -37,13 +61,15 @@ import { useDisplay } from "vuetify";
 import VueCal from "vue-cal";
 import "vue-cal/dist/vuecal.css";
 import DmSectionCard from "../components/sections/DmSectionCard.vue";
+import CalendarMessage from "../components/calendar/DmCalendarMessage.vue";
 
 export default {
   name: "CalendarShow",
 
   components: {
     VueCal,
     DmSectionCard,
+    CalendarMessage,
```
Use the same name for import as the component name.

```javascript
import DmCalendarMessage from "../components/calendar/DmCalendarMessage.vue"
```

BaaltRodrigo commented on src/views/CalendarShow.vue:
```diff
@@ -59,17 +85,22 @@ export default {
     loaded: false,
     sectionInformation: false,
     section: null,
+    dialog: false,
```
Use descriptive names on the variables.

Dialog... what dialog?

BaaltRodrigo reviewed (changes requested):
There are a few things to change. Nothing strictly wrong but that can be improved.

We reach a point when _just works_ is not enough. Try to think that other people needs to understand the code that we are writting.

cristobalwalters12 committed:
Refactor calendar component and add a new save functionality

- If the shared calendar is saved to the user's account, the save button instantly disappears.
- There are no more console logs.
- Some of the code has been refactored.

cristobalwalters12 committed:
Remove unused calendarShared variable
